### PR TITLE
#1095 add support to find element by index

### PR DESCRIPTION
**Pull request by saikrishna321**
Fixes https://github.com/getgauge/taiko/issues/1095

@zabil Can you please review

@msx752

### CODE REVIEW

**Comment by msx752** [comment]
@saikrishna321 it seems good, very useful a function what i need 👍

File: test/unit-tests/$.test.js [hunk 1]
```diff
@@ -117,5 +117,18 @@ describe(test_name, () => {
       const elems = await $('#foo').elements();
       expect(await elems[0].text()).to.be.eql('taiko');
     });
+
+    it('test text() with element index', async () => {
```
Comment (zabil) [inline test/unit-tests/$.test.js]: Can we have a test where there are more than one element?
An id is unique so ideally we should be match by something else.

File: lib/elementWrapper/elementWrapper.js [hunk 2]
```diff
@@ -49,6 +49,18 @@ class ElementWrapper {
...
   }
+
+  async element(index, retryInterval, retryTimeout) {
+    const results = await _getIfExists(this._get, this._description)(
+      null,
+      retryInterval,
+      retryTimeout,
+    );
+    if (index > results.length - 1) {
+      throw new Error(`Element Index is out of range. Found index range ${results.length - 1}`);
```
Comment (zabil) [inline lib/elementWrapper/elementWrapper.js]:
```suggestion
      throw new Error(`Element index is out of range. There are only ${results.length} elements`);
```

File: lib/elementWrapper/elementWrapper.js [hunk 3]
```diff
@@ -49,6 +49,18 @@ class ElementWrapper {
...
   }
+
+  async element(index, retryInterval, retryTimeout) {
+    const results = await _getIfExists(this._get, this._description)(
+      null,
+      retryInterval,
+      retryTimeout,
+    );
+    if (index > results.length - 1) {
+      throw new Error(`Element index is out of range. There are only ${results.length} elements`);
```
Comment (zabil) [inline lib/elementWrapper/elementWrapper.js]: One more small change. I noticed that if there are only 1 element the message can read as follows.

```suggestion
      throw new Error(`Element index is out of range. There are only ${results.length} element(s)`);
```

This way the message will be readable for 1 and more than one elements.

File: test/unit-tests/$.test.js [hunk 4]
```diff
@@ -117,5 +117,18 @@ describe(test_name, () => {
...
+      const elems = await $('#foo').element(0);
+      expect(await elems.text()).to.be.eql('taiko');
+    });
+
+    it('Should throw error when element index is out of bound', async () => {
+      await $('#foo')
+        .element(1)
+        .catch(err => {
+          expect(err).to.match(/Element index is out of range. There are only 1 elements/);
```
Comment (saikrishna321) [inline test/unit-tests/$.test.js]:
```suggestion
          expect(err).to.match(/Element index is out of range. There are only 1 element(s)/);
```

**Pull request merged**